src/DIRAC/WorkloadManagementSystem/Utilities/JobModel.py

```diff
@@ -0,0 +1,192 @@
+        if v not in possibleLogLevels:
+            raise ValueError(f"Log level {v} not in {possibleLogLevels}")
+        return v
+
+    @validator("minNumberOfProcessors")
+    def checkMinNumberOfProcessorsBounds(cls, v):
+        minNumberOfProcessors = Operations().getValue("JobDescription/MinNumberOfProcessors", 1)
+        maxNumberOfProcessors = Operations().getValue("JobDescription/MaxNumberOfProcessors", 1024)
+        if not minNumberOfProcessors <= v <= maxNumberOfProcessors:
+            raise ValueError(
+                f"minNumberOfProcessors out of bounds (must be between {minNumberOfProcessors} and {maxNumberOfProcessors})"
+            )
+        return v
+
+    @validator("maxNumberOfProcessors")
+    def checkMaxNumberOfProcessorsBounds(cls, v):
+        minNumberOfProcessors = Operations().getValue("JobDescription/MinNumberOfProcessors", 1)
+        maxNumberOfProcessors = Operations().getValue("JobDescription/MaxNumberOfProcessors", 1024)
+        if not minNumberOfProcessors <= v <= maxNumberOfProcessors:
+            raise ValueError(
+                f"minNumberOfProcessors out of bounds (must be between {minNumberOfProcessors} and {maxNumberOfProcessors})"
+            )
+        return v
+
+    @root_validator
+    def checkThatMaxNumberOfProcessorsIsGreaterThanMinNumberOfProcessors(cls, values):
+        if values["maxNumberOfProcessors"] and values["minNumberOfProcessors"]:
+            if values["maxNumberOfProcessors"] < values["minNumberOfProcessors"]:
+                raise ValueError("maxNumberOfProcessors must be greater than minNumberOfProcessors")
+        return values
+
+    @root_validator
+    def addTagsDependingOnNumberOfProcessors(cls, values):
+        if values["maxNumberOfProcessors"] and values["minNumberOfProcessors"] == values["maxNumberOfProcessors"]:
+            if values["tags"] is None:
+                values["tags"] = set()
+            values["tags"].add(f"{values['minNumberOfProcessors']}Processors")
+        if values["minNumberOfProcessors"] > 1:
+            if values["tags"] is None:
+                values["tags"] = set()
+            values["tags"].add("MultiProcessor")
+
+        return values
+
+    @root_validator
+    def checkSites(cls, values):
+        if values["sites"] and values["bannedSites"]:
+            values["sites"] = values["sites"] - values["bannedSites"]
+            if not values["sites"]:
+                raise ValueError("sites and bannedSites are mutually exclusive")
+            values["bannedSites"] = None
+        return values
+
+    @validator("platform")
+    def checkPlatform(cls, v: str):
+        if v:
+            res = getDIRACPlatforms()
+            if not res["OK"]:
+                raise ValueError(res["Message"])
+            if v not in res["Value"]:
+                raise ValueError("Invalid platform")
+        return v
+
+    @validator("priority")
+    def checkPriorityBounds(cls, v):
```

Comment:
Sometimes the name of the parameter is `v`, sometimes it is `values`, sometimes there is a type hint, sometimes not. It seems inconsistent, is it expected?